src/cluster.h

```diff
@@ -62,8 +63,11 @@ namespace zim
 
       blob_index_t count() const               { return blob_index_t(offsets.size() - 1); }
 
-      zsize_t getBlobSize(blob_index_t n) const  { return zsize_t(offsets[blob_index_type(n)+1].v
-                                                                - offsets[blob_index_type(n)].v); }
+      zsize_t getBlobSize(blob_index_t n) const
+      {
+          if (blob_index_type(n)+1 >= offsets.size()) throw ZimFileFormatError("blob index out of range");
+          return zsize_t(offsets[blob_index_type(n)+1].v - offsets[blob_index_type(n)].v);
+      }
```

Comment:
This method maybe become a bit to complex to being put in the header.
We may move it to `src/cluster.cpp`.

Reply:
done